src/fsspec_xrootd/xrootd.py

```diff
@@ -43,39 +61,184 @@ def _get_kwargs_from_urls(u: str) -> dict[Any, Any]:
+            return url.path
+        elif type(path) == list:
+            paths = []
+            for item in path:
+                paths.append(client.URL(item).path)
+            return paths
+        else:
+            raise OSError("Strip protocol not given string or list")
 
-        return url.path
+    def mkdir(self, path: str, create_parents: bool = True, **kwargs: Any) -> None:
+        if create_parents:
+            status, n = self._myclient.mkdir(
+                path, MkDirFlags.MAKEPATH, timeout=self.timeout
+            )
+        else:
+            status, n = self._myclient.mkdir(path, timeout=self.timeout)
+        if not status.ok:
+            raise OSError(f"Directory not made properly: {status.message}")
 
-    def ls(self, path: str, detail: bool = True, **kwargs: Any) -> list[Any]:
+    def makedirs(self, path: str, exist_ok: bool = False) -> None:
+        exist = self.exists(path)
+        if exist and exist_ok:
+            return
+        elif exist and not exist_ok:
+            raise OSError("Location already exists and exist_ok arg was set to false")
+        else:
+            status, n = self._myclient.mkdir(
+                path, MkDirFlags.MAKEPATH, timeout=self.timeout
+            )
+            if not status.ok:
+                raise OSError(f"Directory not made properly: {status.message}")
 
-        stats, deets = self._myclient.dirlist(path, DirListFlags.STAT)
+    def rmdir(self, path: str) -> None:
+        status, n = self._myclient.rmdir(path, timeout=self.timeout)
+        if not status.ok:
+            raise OSError(f"Directory not removed properly: {status.message}")
 
-        listing = []
+    def _rm(self, path: str) -> None:
+        status, n = self._myclient.rm(path, timeout=self.timeout)
+        if not status.ok:
+            raise OSError(f"File not removed properly: {status.message}")
 
-        if detail:
-            for item in deets:
-                t = ""
-                if item.statinfo.flags and StatInfoFlags.IS_DIR:
-                    t = "directory"
-                elif item.statinfo.flags and StatInfoFlags.OTHER:
-                    t = "other"
+    def touch(self, path: str, truncate: bool = False, **kwargs: Any) -> None:
+        if truncate or not self.exists(path):
+            status, _ = self._myclient.truncate(path, 0, timeout=self.timeout)
+            if not status.ok:
+                raise OSError(f"File not touched properly: {status.message}")
+        else:
+            status, _ = self._myclient.truncate(
+                path, self.info(path).get("size"), timeout=self.timeout
+            )
+            if not status.ok:
+                raise OSError(f"File not touched properly: {status.message}")
+
+    def modified(self, path: str) -> Any:
+        status, statInfo = self._myclient.stat(path, timeout=self.timeout)
+        return statInfo.modtime
+
+    def sign(self, path: str, expiration: int = 100, **kwargs: Any) -> Any:
```

Comment:
I don't think we need to implement this since as far as I know xrootd does not have signed URL concept. (more info: https://www.altostra.com/blog/aws-s3-presigned-url)

Reply:
I agree